datafusion/physical-plan/src/windows/mod.rs

```diff
@@ -217,6 +217,41 @@ fn get_casted_value(
     }
 }
 
+fn rewrite_null_expr_and_data_type(
```

Comment:
for the naming its little bit abstract however its still good name but to help  contributors understand what this method exactly doing lets add comments.

Having comments method increases chances that method can be reusable in future if the similar case shows up

Reply:
> for the naming its little bit abstract however its still good name but to help contributors understand what this method exactly doing lets add comments.
> 
> Having comments method increases chances that method can be reusable in future if the similar case shows up

💯